src/sources/envoy_als/grpc.rs

```diff
@@ -0,0 +1,136 @@
+        let mut in_stream = request.into_inner();
+        let mut shutdown = self.shutdown.clone();
+        let mut stream_identifier = None;
+
+        loop {
+            tokio::select! {
+                _ = &mut shutdown => break,
+                stream_msg = in_stream.message() => {
+                    if let Err(err) = self.handle_message(stream_msg, &mut stream_identifier).await {
+                        match err {
+                            StreamError::StreamClosed => break,
+                            StreamError::Grpc(status) => {
+                                emit!(GrpcError{error: status.clone()});
+                                return Err(status);
+                            }
+                        }
+                    }
+                },
+            }
+        }
+
+        Ok(Response::new(StreamAccessLogsResponse {}))
+    }
+}
+
+impl Service {
+    async fn handle_message(
+        &self,
+        stream_msg: Result<Option<StreamAccessLogsMessage>, Status>,
+        stream_identifier: &mut Option<Identifier>,
+    ) -> Result<(), StreamError> {
+        match stream_msg {
+            Ok(omsg) => match omsg {
+                Some(msg) => {
+                    self.process_logs(msg, stream_identifier).await?;
+                }
+                None => {
+                    // Envoy gracefully closed the stream
+                    return Err(StreamError::StreamClosed);
+                }
+            },
+            Err(err) => {
+                return Err(StreamError::Grpc(err));
+            }
+        }
+        Ok(())
+    }
+
+    async fn process_logs(
+        &self,
+        msg: StreamAccessLogsMessage,
+        stream_identifier: &mut Option<Identifier>,
+    ) -> Result<(), StreamError> {
+        let mut events = vec![];
+
+        // identifier is only sent on the first message of a stream,
+        // but we want to have it alongside all log events so they can be
+        // identified by the Envoy instance that sent them. To do this,
+        // we store the identifier sent on the stream and tack it on to
+        // future events.
+        if let Some(identifier) = msg.identifier {
+            *stream_identifier = Some(identifier);
+        }
+
+        if let Some(logs_entries) = msg.log_entries {
+            match logs_entries {
+                LogEntries::HttpLogs(http_logs) => {
+                    for l in http_logs.log_entry {
+                        let mut evt = LogEvent::default();
+                        evt.insert("http_log", Value::from(l));
+
+                        if let Some(stream_id) = stream_identifier.clone() {
+                            evt.insert("identifier", Value::from(stream_id));
+                        }
+
+                        events.push(evt);
+                    }
+                }
+                LogEntries::TcpLogs(_tcp_logs) => {
+                    warn!("Received TCP log entry. TCP logs are not yet supported by Vector.");
```

Comment:
This needs to emit an error. You can have a look at the object docker_logs emits [here](https://github.com/vectordotdev/vector/blob/master/src/internal_events/docker_logs.rs#L85-L109). Create something similar in `internal_events` and then call `emit!` to raise the error.

This won't prevent any other events from being processed, but it does update the internal metrics produced by Vector to indicate that something has gone wrong and needs looking into.